### PR TITLE
 Adds request/response mapping for SamrOpenGroup

### PR DESCRIPTION
**Pull request by tosmun-r7**

### CODE REVIEW

File: src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java [hunk 1]
```diff
@@ -22,12 +22,12 @@
...
+import com.hierynomus.protocol.commons.EnumWithValue;
+import com.rapid7.client.dcerpc.RPCException;
 import com.rapid7.client.dcerpc.mssamr.messages.*;
-import com.rapid7.client.dcerpc.mssamr.objects.AliasHandle;
-import com.rapid7.client.dcerpc.mssamr.objects.DomainHandle;
-import com.rapid7.client.dcerpc.mssamr.objects.ServerHandle;
-import com.rapid7.client.dcerpc.mssamr.objects.UserHandle;
+import com.rapid7.client.dcerpc.mssamr.objects.*;
 import com.rapid7.client.dcerpc.objects.ContextHandle;
+import com.rapid7.client.dcerpc.objects.RPC_UNICODE_STRING;
```
Comment (sangdo-lee) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java]: where is using RPC_UNICODE_STRING from this PR?

Comment (tosmun-r7) [reply src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java]: It's not, thanks :)

File: src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java [hunk 2]
```diff
@@ -37,22 +37,21 @@ public SecurityAccountManagerService(RPCTransport transport) {
...
+        final SamrOpenDomainRequest request = new SamrOpenDomainRequest(serverHandle, sid, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
         final SamrOpenDomainResponse response = transport.call(request);
         return response.getHandle();
     }
 
-    public DomainHandle openDomain(ServerHandle serverHandle, SID sid) throws IOException {
-        final SamrOpenDomainRequest request = new SamrOpenDomainRequest(serverHandle, sid, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
-        final SamrOpenDomainResponse response = transport.call(request);
+    public GroupHandle openGroup(DomainHandle domainHandle, int groupRID) throws IOException {
+        final SamrOpenGroupRequest request = new SamrOpenGroupRequest(domainHandle, (int) AccessMask.MAXIMUM_ALLOWED.getValue(), groupRID);
```
Comment (sangdo-lee) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java]: FYI - the legacy code using this `AccessMask (0x00000010)`, is it working when you tested? 
` SamOpenDomainGroupRpcRequest openReq = getSamOpenDomainGroupRpcRequest(
               domHandle,
               cifsSID.getSubAuthorityIDs()[cifsSID.getSubAuthorityCount() - 1],
               0x00000010);`

File: src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java [hunk 3]
```diff
@@ -23,10 +23,7 @@
 import com.hierynomus.msdtyp.AccessMask;
 import com.hierynomus.msdtyp.SID;
 import com.rapid7.client.dcerpc.mssamr.messages.*;
-import com.rapid7.client.dcerpc.mssamr.objects.AliasHandle;
-import com.rapid7.client.dcerpc.mssamr.objects.DomainHandle;
-import com.rapid7.client.dcerpc.mssamr.objects.ServerHandle;
-import com.rapid7.client.dcerpc.mssamr.objects.UserHandle;
+import com.rapid7.client.dcerpc.mssamr.objects.*;
```
Comment (bli-r7) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java]: Revert wildcard imports.

File: src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java [hunk 4]
```diff
@@ -37,22 +34,21 @@ public SecurityAccountManagerService(RPCTransport transport) {
...
     }
 
-    public ServerHandle getServerHandle(String serverName) throws IOException {
+    public ServerHandle openServerHandle(String serverName) throws IOException {
         final SamrConnect2Request request = new SamrConnect2Request(serverName, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
         final SamrConnect2Response response = transport.call(request);
         return response.getHandle();
     }
 
-    public DomainHandle openDomain(String serverName, SID sid) throws IOException {
```
Comment (bli-r7) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java]: I feel removing this is out of scope but it's up to you.

File: src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrOpenGroupRequest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        super(OP_NUM);
+        this.domainHandle = domainHandle;
+        this.desiredAccess = desiredAccess;
+        this.groupRID = groupRID;
+    }
+
+    @Override
+    public void marshal(PacketOutput packetOut) throws IOException {
+        packetOut.writeMarshallable(domainHandle);
+        // No align necessary - Always 24 bytes written
```
Comment (bli-r7) [inline src/main/java/com/rapid7/client/dcerpc/mssamr/messages/SamrOpenGroupRequest.java]: handle is 20 bytes long.

**Pull request merged**